proposals/0023-asynchronous-program-execution-and-broadcast.md

```diff
@@ -0,0 +1,148 @@
+
+The N concurrent Builder have 200ms slots to create blocks out
+of user transactions. These are transmitted to the network.  The
+leader receives and decodes them and generates a UserBlockEntry,
+and adds it to PoH as soon as the leaders PoH has passed the
+UserBlockSlot.
+
+### Fork Choice
+
+If a validator doesn't have the builder's UserBlock, the validator
+doesn't vote on the proposed block and tries to repair the UserBlock.
+That fork is ignored for fork choice.
+
+Otherwise the validator evaluates the forks as usual. 
+
+### UserBlock execution
+
+Each UserBlock is assumed to have been created simultaneously
+during its scheduled slot.  For each UserBlock, the transactions
+are ordered by priority fee before execution. If two transactions
+from two different blocks have the same priority, the earliest
+transaction by UserBlock PoH is executed first, otherwise by which
+UserBlock appears first in the leaders PoH.
+
+### BankHash
+
+Votes must include the most recent computed BankHash along with the
+slot for the BankHash computation.  This slot is different
+from the tower vote slot.
+
+Each validator that is computing a BankHash will be able to verify
+the BankHash values created by block producers for previous slots
+because each BankHash has a link to the previous one.
+
+During verification, if a validator sees that > 1/3 of the network
+has incorrectly computed a state transition, they should halt
+immediately.
+
+### Client Confirmations
+
+Client can receive a confirmation as soon as the UserBlockEntry has
+been optimistically confirmed. Client can also wait for a status
+code weighted by stake weight, as each validator's execution catches
+up to the slot and includes a BankHash confirming the state of the
+execution.
+
+It may be sufficient to wait for 1/3+ validators executing the state
+transition instead of the full 2/3+, because if 1/3+ are incorrect
+the network will halt anyways.
+
+## Impact
+
+Multiple nodes can operate as Builder on the network concurrently.
+So clients can pick the nearest one, and the bandwidth to schedule
+and prioritize transactions is doubled.  There needs to be a design
+for BundleTransactions that allow the bundler to prioritize a batch
+of transactions for execution by paying a priority fee for all of
+them, and executing the whole batch together.
+
+Priority fees now also imply execution priority.
+
+Fork choice and voting is not blocked by user block propagation or
+execution of user transactions.
+
+Builders are going to capture all the MEV.
+
+## Security Considerations
+
+* Validators may include incorrect BankHash values which may not
+be detected instantly. Validators may skip evaluating their own
+BankHash and copy results from other validators.
+
+* Network halts if it cannot compute a epoch snapshot hash once an
+epoch because it is overloaded with user transactions.
+
+* Leader could censor the builder. But then they would miss out
```

Comment:
No, leader can only inject votes and UserBlocks from the scheduled Builders

Reply:
Validators can reject any leader blocks that contain other entries.  

TBD if a signature is necessary, because shreds are signed and validators will verify that the shreds are signed by the builder.

Reply:
I think a signature is necessary, added the spec of it in the UserBlock format section